src/silx/gui/utils/matplotlib.py

```diff
@@ -38,27 +38,113 @@
 __date__ = "02/05/2018"
 
 
+import io
 from pkg_resources import parse_version
 import matplotlib
+import numpy
 
 from .. import qt
 
 
+def rasterMathText(text, font, size=-1, weight=-1, italic=False, devicePixelRatio=1.0):
+    """Raster text using matplotlib supporting latex-like math syntax.
+
+    It supports multiple lines.
+
+    :param str text: The text to raster
+    :param font: Font name or QFont to use
+    :type font: str or :class:`QFont`
+    :param int size:
+        Font size in points
+        Used only if font is given as name.
+    :param int weight:
+        Font weight in [0, 99], see QFont.Weight.
+        Used only if font is given as name.
+    :param bool italic:
+        True for italic font (default: False).
+        Used only if font is given as name.
+    :param float devicePixelRatio:
+        The current ratio between device and device-independent pixel
+        (default: 1.0)
+    :return: Corresponding image in gray scale and baseline offset from top
+    :rtype: (HxW numpy.ndarray of uint8, int)
+    """
+    # Implementation adapted from:
+    # https://github.com/matplotlib/matplotlib/blob/d624571a19aec7c7d4a24123643288fc27db17e7/lib/matplotlib/mathtext.py#L264
+
+    # Lazy import to avoid imports before setting matplotlib's rcParams
+    from matplotlib.font_manager import FontProperties
+    from matplotlib.mathtext import MathTextParser
+    from matplotlib import figure
+
+    dpi = 96  # default
+    qapp = qt.QApplication.instance()
+    if qapp:
+        screen = qapp.primaryScreen()
+        if screen:
+            dpi = screen.logicalDotsPerInchY()
+
+    # Make sure dpi is even, it causes issues with array reshape otherwise
+    dpi = ((dpi * devicePixelRatio) // 2) * 2
+
+    stripped_text = text.strip("\n")
+
+    parser = MathTextParser("path")
+    width, height, depth, _, _ = parser.parse(stripped_text, dpi=dpi)
+    width *= 2
+    height *= 2 * (stripped_text.count("\n") + 1)
+
+    if not isinstance(font, qt.QFont):
+        font = qt.QFont(font, size, weight, italic)
+    prop = FontProperties(
+        family=font.family(),
+        style="italic" if font.italic() else "normal",
+        weight=10 * font.weight(),
+        size=font.pointSize(),
+    )
+
+    fig = figure.Figure(figsize=(width / dpi, height / dpi))
+    fig.text(0, depth / height, stripped_text, fontproperties=prop)
+    with io.BytesIO() as buffer:
+        fig.savefig(buffer, dpi=dpi, format="raw")
+        buffer.seek(0)
+        image = numpy.frombuffer(buffer.read(), dtype=numpy.uint8).reshape(
+            int(height), int(width), 4
+        )
```

Comment:
This is not nice, but it is the way it is done in `matplotlib` (except for the BytesIO buffer):
https://github.com/matplotlib/matplotlib/blob/d624571a19aec7c7d4a24123643288fc27db17e7/lib/matplotlib/mathtext.py#L264-L296